src/main/java/org/quiltmc/enigma_plugin/proposal/SimpleSubtypeFieldNameProposer.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.enigma_plugin.proposal;
+
+import org.quiltmc.enigma.api.Enigma;
+import org.quiltmc.enigma.api.analysis.index.jar.JarIndex;
+import org.quiltmc.enigma.api.translation.mapping.EntryMapping;
+import org.quiltmc.enigma.api.translation.mapping.EntryRemapper;
+import org.quiltmc.enigma.api.translation.representation.entry.ClassEntry;
+import org.quiltmc.enigma.api.translation.representation.entry.Entry;
+import org.quiltmc.enigma.api.translation.representation.entry.FieldEntry;
+import org.quiltmc.enigma.api.translation.representation.entry.LocalVariableEntry;
+import org.quiltmc.enigma_plugin.index.JarIndexer;
+import org.quiltmc.enigma_plugin.index.simple_type_single.SimpleSubtypeSingleIndex;
+import org.quiltmc.enigma_plugin.index.simple_type_single.SimpleSubtypeSingleIndex.FieldInfo;
+import org.quiltmc.enigma_plugin.index.simple_type_single.SimpleSubtypeSingleIndex.SubtypeEntry;
+import org.quiltmc.enigma_plugin.util.StringUtil;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static org.quiltmc.enigma_plugin.util.CasingUtil.toScreamingSnakeCase;
+
+public class SimpleSubtypeFieldNameProposer extends NameProposer {
+	public static final String ID = "simple_subtype_field_names";
+	private final SimpleSubtypeSingleIndex index;
+
+	private static String unCapitalize(String string) {
```

Comment:
why not just use toLowerCase(Local.ROOT)? i imagine there's slightly less overhead here, but it's so negligeable

Reply:
`unCapitalize` only converts the first character, `toLowerCase` converts all characters
`unCapitalize` is for converting from `UpperCamelCase` to `lowerCamelCase`

Reply:
👍 oops